test/test_qdwh.cc

```diff
@@ -0,0 +1,244 @@
+        H_data.resize( lldH * nlocH );
+        H = slate::Matrix<scalar_t>::fromScaLAPACK(n, n, &H_data[0], lldH, nb, p, q, MPI_COMM_WORLD);
+    }
+
+    real_t cond = 1 / std::numeric_limits<real_t>::epsilon();
+    params.matrix.kind.set_default("svd");
+    params.matrix.cond.set_default(cond);
+
+    slate::generate_matrix( params.matrix, A);
+
+    // if check is required, copy test data
+    slate::Matrix<scalar_t> Aref;
+    std::vector<scalar_t> Aref_data;
+    if (check || ref) {
+        Aref_data.resize( lldA * nlocA );
+        Aref = slate::Matrix<scalar_t>::fromScaLAPACK(
+            m, n, &Aref_data[0], lldA, nb, p, q, MPI_COMM_WORLD);
+        slate::copy(A, Aref);
+    }
+
+    print_matrix("A", A, params);
+
+    // todo: how to compute gflops?
+
+    if (! ref_only) {
+        if (trace) slate::trace::Trace::on();
+        else slate::trace::Trace::off();
+
+        int itqr = 0, itpo = 0;
+
+        double time = barrier_get_wtime(MPI_COMM_WORLD);
+
+        //==================================================
+        // Run SLATE test.
+        //==================================================
+        // A = AH,
+        // A will be written by the orthogonal polar factor
+        // H is the symmetric positive semidefinite polar factor
+        slate::qdwh(A, H, itqr, itpo, opts);
+
+        time = barrier_get_wtime(MPI_COMM_WORLD) - time;
+
+        // gflop include the flops to estimate the condition
+        // number, the flops for the qr_based iterations the
+        // flops for the po_based iterations and the flops
+        // to compute H (hermition polar factor).
+        // gflop = trcon + itqr * (flop_per_qr_based_iter) +
+        //         itpo * (flop_per_po_based_iter) +
+        //         flop_compute_H
+        double gflop_trcon = lapack::Gflop<scalar_t>::geqrf(m, n);
+
+        // this flop count of the QR-based iteration is in case we are not taking advantage
+        // of the identity trailing submatrix:
+        //double gflop_itqr  = itqr * ( lapack::Gflop<scalar_t>::geqrf(m+n, n) +
+        //                              lapack::Gflop<scalar_t>::unmqr(slate::Side::Left, m+n, n, n) +
+        //                              blas::Gflop<scalar_t>::gemm(m, n, n) );
+        // when take advantage of the matrix structure, the flops count will be reduced:
+        double gflop_itqr  = itqr * ( 1e-9 * 2. * double(m*n*n) +
+                                      1e-9 * 2. * double(m*n*n) +
+                                      blas::Gflop<scalar_t>::gemm(m, n, n) );
+
+        double gflop_itpo  = itpo * ( blas::Gflop<scalar_t>::gemm(m, n, n) +
+                                      lapack::Gflop<scalar_t>::potrf(m)      +
+                                      blas::Gflop<scalar_t>::trsm(slate::Side::Left, m, n) );
+
+        double gflop_compute_H = blas::Gflop<scalar_t>::her2k(n, m);
```

Comment:
Currently, this is really `gemm`, but eventually it should be `herk` (i.e., `herkx`) instead of `her2k`.

(I'll fix.)